packages/flutter/test/material/navigation_bar_test.dart

```diff
@@ -0,0 +1,429 @@
+
+void main() {
+  setUp(() async {
+    WidgetsBinding.instance!.focusManager.highlightStrategy =
+        FocusHighlightStrategy.alwaysTraditional;
+  });
+
+  testWidgets('Navigation bar updates destinations when tapped', (WidgetTester tester) async {
+    int mutatedIndex = 0;
+    final Widget widget = _buildWidget(
+      UniqueKey(),
+      NavigationBar(
+        destinations: const <Widget>[
+          NavigationDestination(
+            icon: Icon(Icons.ac_unit),
+            label: 'AC',
+          ),
+          NavigationDestination(
+            icon: Icon(Icons.access_alarm),
+            label: 'Alarm',
+          ),
+        ],
+        onDestinationSelected: (int i) {
+          mutatedIndex = i;
+        },
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+    await tester.tap(find.text('Alarm'));
+
+    expect(mutatedIndex, 1);
+  });
+
+  testWidgets('NavigationBar content test', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    expect(find.text('AC'), findsOneWidget);
+    expect(find.text('Alarm'), findsOneWidget);
+  });
+
+  testWidgets('NavigationBar can update background color', (WidgetTester tester) async {
+    const Color color = Colors.yellow;
+
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          backgroundColor: color,
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    expect(_getMaterial(tester).color, equals(color));
+  });
+
+  testWidgets('NavigationBar adds bottom padding to height', (WidgetTester tester) async {
+    const double bottomPadding = 40.0;
+
+    await tester.pumpWidget(
+      _buildWidget(
+        UniqueKey(),
+        NavigationBar(
+          destinations: const <Widget>[
+            NavigationDestination(
+              icon: Icon(Icons.ac_unit),
+              label: 'AC',
+            ),
+            NavigationDestination(
+              icon: Icon(Icons.access_alarm),
+              label: 'Alarm',
+            ),
+          ],
+          onDestinationSelected: (int i) {},
+        ),
+      ),
+    );
+
+    final double defaultSize = tester.getSize(find.byType(NavigationBar)).height;
```

Comment:
This is guaranteed to be 80 (?) per the docs. Shouldn't we check that?

Reply:
Done